### PR TITLE
fix(textbox): fix BeforeTextChanging cancelling on Wpf

### LINKED ISSUE #15372: [Skia] TextBox: Cancel BeforeTextChanging moves cursor in front of text
### Current behavior

Cancelling BeforeTextChanging moves cursor in front of text

### Expected behavior

Cursor stays on current position

### How to reproduce it (as minimally and precisely as possible)

Add the handler to a textbox and dismiss the new input

E.g:
```
        public void TextBox_BeforeTextChanging(TextBox sender, TextBoxBeforeTextChangingEventArgs args)
        {
            args.Cancel = !double.TryParse(args.NewText, out double _);
        }
```

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

5.1.31

### Affected platforms

Skia (WPF)

### IDE

Visual Studio 2022

### IDE version

17.8

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #15372

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs [hunk 1]
```diff
@@ -168,11 +169,26 @@ internal void UpdateTextFromNative(string newText)
 			var textBox = _textBox?.GetTarget();
 			if (textBox != null)
 			{
-				var text = textBox.ProcessTextInput(newText);
-				SetDisplayBlockText(text);
-				if (text != newText)
+				var oldText = textBox.Text; // preexisting text
+				var oldSelection = SelectionBeforeKeyDown; // On Gtk, SelectionBeforeKeyDown just points to Selection, which is updated by SetTextNative, so we need to read it before SetTextNative.
+				var modifiedText = textBox.ProcessTextInput(newText); // new text after BeforeTextChanging, TextChanging, DP callback, etc
```
[the remark this answers is not in the document]
Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs:174]: We seem to already have `TextBox_BeforeTextChanging`, among others.

[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs:174]: Great, then let's flag this particular test as manual and we need to explain the steps to make in order for the tester to validate the behavior.

[the remark this answers is not in the document]
Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs:174]: The existing test turned out to be too general. I wrote a new test with specific instructions

[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs:174]: Did you push it? :)

[the remark this answers is not in the document]
Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/TextBox/TextBoxView.skia.cs:174]: Apparently not, my bad.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**